test/integration/export-intent/test/index.test.js

```diff
@@ -167,7 +167,7 @@ describe('Application Export Intent Output', () => {
 
         it('should build and export', async () => {
           const result = await nextBuild(appDir, [], { stderr: true })
-          expect(result.stderr).toMatch('A.getInitialProps()')
+          expect(result.stderr).toMatch('.getInitialProps()')
```

Comment:
I found that https://github.com/vercel/next.js/pull/57085 added this, but this should fail on the production builds. We should **not** preserve class names for production (10% size regression if we do so).

Reply:
Due to a regression I introduced before #57085 it was not detected, and this PR includes fixes for the regression.

Reply:
cc @styfle